pkg/collector/tcp.go

```diff
@@ -26,23 +26,22 @@ func (cp *CollectingProcess) startTCPServer() {
 			return
 		}
 		cp.updateAddress(listener.Addr())
-		klog.Infof("Started TLS collecting process on %s", cp.address)
+		klog.Infof("Started TLS collecting process on %s", cp.netAddress)
 	} else {
 		listener, err = net.Listen("tcp", cp.address)
 		if err != nil {
 			klog.Errorf("Cannot start collecting process on %s: %v", cp.address, err)
 			return
 		}
 		cp.updateAddress(listener.Addr())
-		klog.Infof("Start TCP collecting process on %s", cp.address)
+		klog.Infof("Start TCP collecting process on %s", cp.netAddress)
 	}
-
+	defer listener.Close()
 	go func() {
```

Comment:
Lets have some logic to exit go routine properly. What do you think?

```
go func(stopCh chan bool) {
	for {
		conn, err := listener.Accept()
		if err != nil {
		     select {
			case <-stopCh:
				return
			default:
				klog.Errorf("Cannot start the connection on the collecting process at %s: %v", cp.address, err)
					return}
		}
		go cp.handleTCPClient(conn)
	}
}(cp.stopChan)
```

Reply:
With this change, we will need to send multiple signal to stopChan to stop both this goroutine and unblock the rest statements of the function. Do you think that is necessary?

Reply:
You are right. It will be easier if it's not a bool channel.
Could we define the channel as empty strict and close the channel to handle multiple go routines like in a conventional way?